### PR TITLE
Convert BlazorExtension to SDK style project

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes https://github.com/aspnet/AspNetCore/issues/7135

### CODE REVIEW

**Comment by pranavkm** [comment]
The vsixmanifest has the right version - `<Identity Id="Microsoft.VisualStudio.BlazorExtension" Version="16.0.424242.424242" Language="en-US" Publisher="Microsoft" />` and the assembly metadata looks right:

```
[assembly: CompilationRelaxations(8)]
[assembly: RuntimeCompatibility]
[assembly: Debuggable()]
[assembly: TargetFramework(".NETFramework,Version=v4.7.2")]
[assembly: AssemblyMetadata("BuildNumber", "t000")]
[assembly: AssemblyMetadata("CommitHash", "5e5904dc6571c8f7bbc4ab9d5f1274ec40520195")]
[assembly: AssemblyMetadata("SourceCommitUrl", "https://github.com/aspnet/AspNetCore/tree/5e5904dc6571c8f7bbc4ab9d5f1274ec40520195")]
[assembly: AssemblyMetadata("Serviceable", "True")]
[assembly: AssemblyCompany("Microsoft Corporation.")]
[assembly: AssemblyConfiguration("Debug")]
[assembly: AssemblyCopyright("© Microsoft Corporation. All rights reserved.")]
[assembly: AssemblyFileVersion("0.8.0.42424")]
[assembly: AssemblyInformationalVersion("16.0.424242.424242")]
[assembly: AssemblyProduct("Microsoft ASP.NET Core")]
[assembly: AssemblyTitle("Microsoft.VisualStudio.BlazorExtension")]
[assembly: NeutralResourcesLanguage("en-US")]
[assembly: AssemblyVersion("0.8.0.0")]
```

Is there anything else that I'm missing here?

**Comment by natemcmaster** [comment]
Does debugging work in VS 2019? I was under the impression the VSIX tooling didn't support this project style yet.

**Comment by pranavkm** [comment]
@natemcmaster made some tweaks, which makes Ctrl-F5 \ debugging work.

**Comment by natemcmaster** [comment]
Nice 😎

File: src/Components/Blazor/BlazorExtension/src/Microsoft.VisualStudio.BlazorExtension.csproj [hunk 1]
```diff
@@ -1,23 +1,35 @@
...
+
+    <!--
+      Mark the VSIX as a per-computer install (not-per-user). Putting a component "in the box"
+      requires this, and trying to change it after doing a release has lot of problems.
+    -->
+    <ExtensionInstallationRoot>CommonExtensions</ExtensionInstallationRoot>
+    <ExtensionInstallationFolder>Microsoft\Blazor</ExtensionInstallationFolder>
+
+    <!-- This should be set as true if we're ever building the VSIX for inclusion by the VS installer. -->
+    <IsProductComponent Condition="'$(IsProductComponent)'==''">false</IsProductComponent>
```
Comment (natemcmaster) [inline src/Components/Blazor/BlazorExtension/src/Microsoft.VisualStudio.BlazorExtension.csproj]:
```suggestion
    <IsProductComponent>false</IsProductComponent>
```

Removing the condition so this isn't accidentally overridden some how.

File: src/Components/Blazor/BlazorExtension/src/Microsoft.VisualStudio.BlazorExtension.csproj [hunk 2]
```diff
@@ -101,174 +98,50 @@
...
-    <OutputType>Library</OutputType>
-    <AppDesignerFolder>Properties</AppDesignerFolder>
-    <RootNamespace>Microsoft.VisualStudio.BlazorExtension</RootNamespace>
-    <AssemblyName>Microsoft.VisualStudio.BlazorExtension</AssemblyName>
-    <TargetFrameworkVersion>v4.7.2</TargetFrameworkVersion>
-    <GeneratePkgDefFile>true</GeneratePkgDefFile>
-  </PropertyGroup>
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
-    <DebugSymbols>true</DebugSymbols>
-    <DebugType>full</DebugType>
```
Comment (natemcmaster) [inline src/Components/Blazor/BlazorExtension/src/Microsoft.VisualStudio.BlazorExtension.csproj]: Not 100% sure, but I think this Microsoft.NET.Sdk sets this to portable. Not sure if that's important or not.

File: src/Components/Blazor/BlazorExtension/src/Microsoft.VisualStudio.BlazorExtension.csproj [hunk 3]
```diff
@@ -101,174 +98,50 @@
...
-  <PropertyGroup>
-    <SchemaVersion>2.0</SchemaVersion>
-    <ProjectTypeGuids>{82b43b9b-a64c-4715-b499-d71e9ca2bd60};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
-    <ProjectGuid>{9088E4E4-B855-457F-AE9E-D86709A5E1F4}</ProjectGuid>
-    <OutputType>Library</OutputType>
-    <AppDesignerFolder>Properties</AppDesignerFolder>
-    <RootNamespace>Microsoft.VisualStudio.BlazorExtension</RootNamespace>
-    <AssemblyName>Microsoft.VisualStudio.BlazorExtension</AssemblyName>
-    <TargetFrameworkVersion>v4.7.2</TargetFrameworkVersion>
-    <GeneratePkgDefFile>true</GeneratePkgDefFile>
```
Comment (natemcmaster) [inline src/Components/Blazor/BlazorExtension/src/Microsoft.VisualStudio.BlazorExtension.csproj]: Intentionally removed? Seems important.

**Review by natemcmaster — APPROVED** [review]
I don't know as much @rynowak about how VSIX stuff is supposed to work. Added a few minor comments, but as long as it works. :shipit:

File: src/Components/Blazor/BlazorExtension/src/Microsoft.VisualStudio.BlazorExtension.csproj (hunk 2 above)
Comment (pranavkm) [reply src/Components/Blazor/BlazorExtension/src/Microsoft.VisualStudio.BlazorExtension.csproj]: Checked the Razor vsix, and it produces portable pdbs:

image

Guess it doesn't matter.

**Pull request merged**